From ece814cded44c56bfb8d005376f8d999276bf022 Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Mon, 27 Oct 2025 10:31:28 +0100
Subject: [PATCH] Disable virtual vibration support

UN1CA: this patch is not complete! Please check unica/patches/product_feature/customize.sh for more info.

Change Id: 8cd98fd2012d73ce84f7c81af4cde187a5251727
From: Erick-04 <solonsosua@gmail.com>
Date: Sat, 3 Oct 2026 15:57:18

---
 .../server/audio/AudioDeviceBroker.smali      |  10 --
 smali/com/android/server/audio/BtHelper.smali |   4 -
 .../vibrator/VibrationCustomSettings.smali    |   2 -
 .../server/vibrator/VibrationSettings.smali   |   4 +-
 .../vibrator/VibratorManagerService$1.smali   |  75 ---------
 ...nagerService$ExternalVibratorService.smali |  45 ------
 .../vibrator/VibratorManagerService.smali     | 151 +-----------------
 7 files changed, 2 insertions(+), 289 deletions(-)

diff --git a/smali/com/android/server/audio/AudioDeviceBroker.smali b/smali/com/android/server/audio/AudioDeviceBroker.smali
index e1f8cf2..aec16fa 100644
--- a/smali/com/android/server/audio/AudioDeviceBroker.smali
+++ b/smali/com/android/server/audio/AudioDeviceBroker.smali
@@ -771,16 +771,6 @@
 
     invoke-virtual {v0, p1, v4}, Lcom/android/server/audio/AudioDeviceBroker;->onSetBtScoActiveDevice(Landroid/bluetooth/BluetoothDevice;Z)V
 
-    new-instance p1, Lcom/android/server/audio/BtHelper$$ExternalSyntheticLambda0;
-
-    invoke-direct {p1}, Ljava/lang/Object;-><init>()V
-
-    iput-object p0, p1, Lcom/android/server/audio/BtHelper$$ExternalSyntheticLambda0;->f$0:Lcom/android/server/audio/BtHelper;
-
-    invoke-static {}, Ljava/lang/invoke/VarHandle;->storeStoreFence()V
-
-    invoke-static {p1}, Lcom/samsung/android/server/audio/AudioExecutor;->execute(Ljava/lang/Runnable;)V
-
     goto/16 :goto_3
 
     :cond_2
diff --git a/smali/com/android/server/audio/BtHelper.smali b/smali/com/android/server/audio/BtHelper.smali
index 99ea139..0aaa6af 100644
--- a/smali/com/android/server/audio/BtHelper.smali
+++ b/smali/com/android/server/audio/BtHelper.smali
@@ -40,8 +40,6 @@
 
 .field public mScoConnectionState:I
 
-.field public mVibratorManagerInternal:Lcom/android/server/vibrator/VibratorManagerInternal;
-
 
 # direct methods
 .method public static btHeadsetDeviceToAudioDevice(Landroid/bluetooth/BluetoothDevice;)Landroid/media/AudioDeviceAttributes;
@@ -3221,8 +3219,6 @@
     move-result-object v0
 
     check-cast v0, Lcom/android/server/vibrator/VibratorManagerInternal;
-
-    iput-object v0, p0, Lcom/android/server/audio/BtHelper;->mVibratorManagerInternal:Lcom/android/server/vibrator/VibratorManagerInternal;
     :try_end_0
     .catchall {:try_start_0 .. :try_end_0} :catchall_0
 
diff --git a/smali_classes2/com/android/server/vibrator/VibrationCustomSettings.smali b/smali_classes2/com/android/server/vibrator/VibrationCustomSettings.smali
index c854bb6..5a25f53 100644
--- a/smali_classes2/com/android/server/vibrator/VibrationCustomSettings.smali
+++ b/smali_classes2/com/android/server/vibrator/VibrationCustomSettings.smali
@@ -8,8 +8,6 @@
 
 .field public LEVEL_TO_TOUCH_MAGNITUDE:[I
 
-.field public mAudioManager:Landroid/media/AudioManager;
-
 .field public mAudioMode:I
 
 .field public mCallMagnitude:I
diff --git a/smali_classes2/com/android/server/vibrator/VibrationSettings.smali b/smali_classes2/com/android/server/vibrator/VibrationSettings.smali
index 3ae3b49..393ba70 100644
--- a/smali_classes2/com/android/server/vibrator/VibrationSettings.smali
+++ b/smali_classes2/com/android/server/vibrator/VibrationSettings.smali
@@ -1423,11 +1423,9 @@
 
     monitor-enter v8
 
-    :try_start_4
-    iput-object v6, v1, Lcom/android/server/vibrator/VibrationCustomSettings;->mAudioManager:Landroid/media/AudioManager;
-
     if-eqz v6, :cond_3
 
+    :try_start_4
     iget-object v7, v1, Lcom/android/server/vibrator/VibrationCustomSettings;->mContext:Landroid/content/Context;
 
     invoke-virtual {v7}, Landroid/content/Context;->getMainExecutor()Ljava/util/concurrent/Executor;
diff --git a/smali_classes2/com/android/server/vibrator/VibratorManagerService$1.smali b/smali_classes2/com/android/server/vibrator/VibratorManagerService$1.smali
index b9c79b2..b68e9c3 100644
--- a/smali_classes2/com/android/server/vibrator/VibratorManagerService$1.smali
+++ b/smali_classes2/com/android/server/vibrator/VibratorManagerService$1.smali
@@ -193,81 +193,6 @@
     goto :goto_0
 
     :cond_3
-    const-string/jumbo p1, "android.media.STREAM_DEVICES_CHANGED_ACTION"
-
-    invoke-virtual {p1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result p1
-
-    const/4 v3, 0x2
-
-    if-eqz p1, :cond_4
-
-    const-string/jumbo p1, "android.media.EXTRA_VOLUME_STREAM_TYPE"
-
-    invoke-virtual {p2, p1, v1}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
-
-    move-result p1
-
-    const-string/jumbo v0, "android.media.EXTRA_VOLUME_STREAM_DEVICES"
-
-    invoke-virtual {p2, v0, v1}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
-
-    move-result p2
-
-    const/4 v0, 0x3
-
-    if-ne p1, v0, :cond_8
-
-    and-int/lit8 p1, p2, 0x2
-
-    if-nez p1, :cond_8
-
-    const-string p1, "Detected device changed"
-
-    invoke-virtual {p0, p1}, Lcom/android/server/vibrator/VibratorManagerService$1;->externalDeviceConnected(Ljava/lang/String;)V
-
-    goto :goto_1
-
-    :cond_4
-    const-string/jumbo p1, "android.intent.action.HEADSET_PLUG"
-
-    invoke-virtual {p1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result p1
-
-    if-eqz p1, :cond_5
-
-    const-string/jumbo p1, "headset plug"
-
-    invoke-virtual {p0, p1}, Lcom/android/server/vibrator/VibratorManagerService$1;->externalDeviceConnected(Ljava/lang/String;)V
-
-    goto :goto_1
-
-    :cond_5
-    const-string/jumbo p1, "android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED"
-
-    invoke-virtual {p1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
-
-    move-result p1
-
-    if-eqz p1, :cond_6
-
-    const-string/jumbo p1, "android.bluetooth.profile.extra.STATE"
-
-    invoke-virtual {p2, p1, v0}, Landroid/content/Intent;->getIntExtra(Ljava/lang/String;I)I
-
-    move-result p1
-
-    if-ne p1, v3, :cond_8
-
-    const-string/jumbo p1, "a2dp device connected"
-
-    invoke-virtual {p0, p1}, Lcom/android/server/vibrator/VibratorManagerService$1;->externalDeviceConnected(Ljava/lang/String;)V
-
-    goto :goto_1
-
-    :cond_6
     const-string/jumbo p1, "com.android.launcher3.quickstep.closeall"
 
     invoke-virtual {p1, v2}, Ljava/lang/String;->equals(Ljava/lang/Object;)Z
diff --git a/smali_classes2/com/android/server/vibrator/VibratorManagerService$ExternalVibratorService.smali b/smali_classes2/com/android/server/vibrator/VibratorManagerService$ExternalVibratorService.smali
index 2a479a0..d22ce22 100644
--- a/smali_classes2/com/android/server/vibrator/VibratorManagerService$ExternalVibratorService.smali
+++ b/smali_classes2/com/android/server/vibrator/VibratorManagerService$ExternalVibratorService.smali
@@ -344,16 +344,6 @@
 
     invoke-interface {v5, v6, v7, v9}, Lcom/android/server/vibrator/VibrationSession;->requestEnd(Lcom/android/server/vibrator/VibrationSession$Status;Lcom/android/server/vibrator/VibrationSession$CallerInfo;Z)V
 
-    iget-object v5, p0, Lcom/android/server/vibrator/VibratorManagerService$ExternalVibratorService;->this$0:Lcom/android/server/vibrator/VibratorManagerService;
-
-    iget-object v5, v5, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz v5, :cond_5
-
-    const-string/jumbo v6, "cancel"
-
-    invoke-virtual {v5, v6}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
-
     :cond_5
     move v7, v9
 
@@ -602,33 +592,6 @@
     invoke-virtual {v4, v0}, Lcom/android/server/vibrator/VibratorManagerService;->setAchMagnitude(Lcom/android/server/vibrator/ExternalVibrationSession;)V
 
     :cond_b
-    iget-object v4, v0, Lcom/android/server/vibrator/ExternalVibrationSession;->mScale:Landroid/os/ExternalVibrationScale;
-
-    iget v4, v4, Landroid/os/ExternalVibrationScale;->scaleLevel:I
-
-    if-eq v4, v3, :cond_c
-
-    iget-object v3, p0, Lcom/android/server/vibrator/VibratorManagerService$ExternalVibratorService;->this$0:Lcom/android/server/vibrator/VibratorManagerService;
-
-    iget-object v3, v3, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz v3, :cond_c
-
-    invoke-virtual {p1}, Landroid/os/ExternalVibration;->getPackage()Ljava/lang/String;
-
-    move-result-object v4
-
-    invoke-virtual {p1}, Landroid/os/ExternalVibration;->getVibrationAttributesWithTags()Landroid/os/VibrationAttributes;
-
-    move-result-object v6
-
-    invoke-virtual {p1}, Landroid/os/ExternalVibration;->isRepeating()Z
-
-    move-result v7
-
-    invoke-virtual {v3, v4, v6, v7}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->playVirtualSoundIfNeeded(Ljava/lang/String;Landroid/os/VibrationAttributes;Z)V
-
-    :cond_c
     iget-object v3, p0, Lcom/android/server/vibrator/VibratorManagerService$ExternalVibratorService;->this$0:Lcom/android/server/vibrator/VibratorManagerService;
 
     invoke-virtual {v3, p1}, Lcom/android/server/vibrator/VibratorManagerService;->reportVibrationInfo(Landroid/os/ExternalVibration;)V
@@ -814,14 +777,6 @@
 
     iget-object p1, p0, Lcom/android/server/vibrator/VibratorManagerService$ExternalVibratorService;->this$0:Lcom/android/server/vibrator/VibratorManagerService;
 
-    iget-object p1, p1, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p1, :cond_0
-
-    const-string/jumbo v0, "stop"
-
-    invoke-virtual {p1, v0}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
-
     goto :goto_0
 
     :catchall_0
diff --git a/smali_classes2/com/android/server/vibrator/VibratorManagerService.smali b/smali_classes2/com/android/server/vibrator/VibratorManagerService.smali
index d9eb81d..dea2be6 100644
--- a/smali_classes2/com/android/server/vibrator/VibratorManagerService.smali
+++ b/smali_classes2/com/android/server/vibrator/VibratorManagerService.smali
@@ -106,8 +106,6 @@
 
 .field public final mVibratorManagerRecords:Lcom/android/server/vibrator/VibratorManagerService$VibratorManagerRecords;
 
-.field public mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
 .field public final mWakeLock:Landroid/os/PowerManager$WakeLock;
 
 
@@ -1570,88 +1568,6 @@
 
     invoke-virtual {v0}, Ljava/lang/Object;->getClass()Ljava/lang/Class;
 
-    new-instance v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    iget-object v1, p0, Lcom/android/server/vibrator/VibratorManagerService;->mContext:Landroid/content/Context;
-
-    iget-object v2, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVibrationSettings:Lcom/android/server/vibrator/VibrationSettings;
-
-    invoke-direct {v0}, Ljava/lang/Object;-><init>()V
-
-    iput-object v2, v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;->mSettings:Lcom/android/server/vibrator/VibrationSettings;
-
-    invoke-static {v1}, Lcom/samsung/android/server/audio/AudioSettingsHelper;->getInstance(Landroid/content/Context;)Lcom/samsung/android/server/audio/AudioSettingsHelper;
-
-    move-result-object v1
-
-    iput-object v1, v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;->mAudioSettingsHelper:Lcom/samsung/android/server/audio/AudioSettingsHelper;
-
-    new-instance v1, Landroid/media/SoundPool$Builder;
-
-    invoke-direct {v1}, Landroid/media/SoundPool$Builder;-><init>()V
-
-    const/4 v2, 0x1
-
-    invoke-virtual {v1, v2}, Landroid/media/SoundPool$Builder;->setMaxStreams(I)Landroid/media/SoundPool$Builder;
-
-    move-result-object v1
-
-    new-instance v3, Landroid/media/AudioAttributes$Builder;
-
-    invoke-direct {v3}, Landroid/media/AudioAttributes$Builder;-><init>()V
-
-    const/16 v4, 0xd
-
-    invoke-virtual {v3, v4}, Landroid/media/AudioAttributes$Builder;->setUsage(I)Landroid/media/AudioAttributes$Builder;
-
-    move-result-object v3
-
-    invoke-virtual {v3, v2}, Landroid/media/AudioAttributes$Builder;->setFlags(I)Landroid/media/AudioAttributes$Builder;
-
-    move-result-object v2
-
-    const/4 v3, 0x4
-
-    invoke-virtual {v2, v3}, Landroid/media/AudioAttributes$Builder;->setContentType(I)Landroid/media/AudioAttributes$Builder;
-
-    move-result-object v2
-
-    const-string/jumbo v3, "VIRTUAL_VIB_SOUND"
-
-    invoke-virtual {v2, v3}, Landroid/media/AudioAttributes$Builder;->semAddAudioTag(Ljava/lang/String;)Landroid/media/AudioAttributes$Builder;
-
-    move-result-object v2
-
-    invoke-virtual {v2}, Landroid/media/AudioAttributes$Builder;->build()Landroid/media/AudioAttributes;
-
-    move-result-object v2
-
-    invoke-virtual {v1, v2}, Landroid/media/SoundPool$Builder;->setAudioAttributes(Landroid/media/AudioAttributes;)Landroid/media/SoundPool$Builder;
-
-    move-result-object v1
-
-    invoke-virtual {v1}, Landroid/media/SoundPool$Builder;->build()Landroid/media/SoundPool;
-
-    move-result-object v1
-
-    iput-object v1, v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;->mSoundPool:Landroid/media/SoundPool;
-
-    const/4 v2, 0x0
-
-    iput-boolean v2, v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;->mIsVirtualSoundPlaying:Z
-
-    const-string v3, "/system/media/audio/ui/VIB_Vibration_Call.ogg"
-
-    invoke-virtual {v1, v3, v2}, Landroid/media/SoundPool;->load(Ljava/lang/String;I)I
-
-    move-result v1
-
-    iput v1, v0, Lcom/android/server/vibrator/VirtualVibSoundHelper;->mVirtualSoundId:I
-
-    invoke-static {}, Ljava/lang/invoke/VarHandle;->storeStoreFence()V
-
-    iput-object v0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
     invoke-virtual {p0}, Lcom/android/server/vibrator/VibratorManagerService;->registerFoldStateListener()V
 
     return-void
@@ -1679,15 +1595,6 @@
 
     :cond_1
     :goto_0
-    iget-object p0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p0, :cond_2
-
-    const-string/jumbo p1, "recovery"
-
-    invoke-virtual {p0, p1}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
-
-    :cond_2
     return-void
 .end method
 
@@ -1825,13 +1732,6 @@
 
     invoke-interface {p1, p2}, Lcom/android/server/vibrator/VibrationSession;->requestEnd(Lcom/android/server/vibrator/VibrationSession$Status;)V
 
-    iget-object p0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p0, :cond_4
-
-    const-string/jumbo p1, "cancel"
-
-    invoke-virtual {p0, p1}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
     :try_end_4
     .catchall {:try_start_4 .. :try_end_4} :catchall_0
 
@@ -7052,7 +6952,7 @@
 
     const-string/jumbo v0, "com.sec.media.action.VIBRTOR_LOGGING"
 
-    const-string/jumbo v1, "android.media.STREAM_DEVICES_CHANGED_ACTION"
+    const-string/jumbo v1, "com.android.launcher3.quickstep.closeall"
 
     const-string/jumbo v2, "com.android.server.sepunion.semgoodcatchservice.GOOD_CATCH_STATE_CHANGED"
 
@@ -7064,18 +6964,6 @@
 
     move-result-object v0
 
-    const-string/jumbo v1, "android.intent.action.HEADSET_PLUG"
-
-    invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
-
-    const-string/jumbo v1, "android.bluetooth.a2dp.profile.action.CONNECTION_STATE_CHANGED"
-
-    invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
-
-    const-string/jumbo v1, "com.android.launcher3.quickstep.closeall"
-
-    invoke-virtual {v0, v1}, Landroid/content/IntentFilter;->addAction(Ljava/lang/String;)V
-
     iget-object p0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mSamsungReceiver:Landroid/content/BroadcastReceiver;
 
     const/4 v1, 0x4
@@ -9622,13 +9510,6 @@
     move-result-object p6
 
     :goto_2
-    iget-object p1, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p1, :cond_b
-
-    const-string/jumbo p7, "cancel"
-
-    invoke-virtual {p1, p7}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
     :try_end_1
     .catchall {:try_start_1 .. :try_end_1} :catchall_1
 
@@ -10008,20 +9889,6 @@
     throw p0
 
     :cond_2
-    iget-object p0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p0, :cond_3
-
-    iget-object v1, v2, Lcom/android/server/vibrator/VibrationSession$CallerInfo;->opPkg:Ljava/lang/String;
-
-    iget-object v2, v2, Lcom/android/server/vibrator/VibrationSession$CallerInfo;->attrs:Landroid/os/VibrationAttributes;
-
-    invoke-virtual {p1}, Lcom/android/server/vibrator/SingleVibrationSession;->isRepeating()Z
-
-    move-result p1
-
-    invoke-virtual {p0, v1, v2, p1}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->playVirtualSoundIfNeeded(Ljava/lang/String;Landroid/os/VibrationAttributes;Z)V
-
     :cond_3
     return-object v0
 
@@ -10502,14 +10369,6 @@
 
     invoke-interface {v1, v2}, Lcom/android/server/vibrator/VibrationSession;->requestEnd(Lcom/android/server/vibrator/VibrationSession$Status;)V
 
-    iget-object p0, p0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz p0, :cond_4
-
-    const-string/jumbo v1, "cancel"
-
-    invoke-virtual {p0, v1}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
-
     :cond_4
     monitor-exit v0
 
@@ -11150,14 +11009,6 @@
 
     :cond_f
     :goto_5
-    iget-object v3, v0, Lcom/android/server/vibrator/VibratorManagerService;->mVirtualVibSoundHelper:Lcom/android/server/vibrator/VirtualVibSoundHelper;
-
-    if-eqz v3, :cond_10
-
-    const-string/jumbo v4, "cancel"
-
-    invoke-virtual {v3, v4}, Lcom/android/server/vibrator/VirtualVibSoundHelper;->stopVirtualSound(Ljava/lang/String;)V
-
     :cond_10
     sget-object v3, Lcom/android/server/vibrator/VibrationSession$Status;->CANCELLED_SUPERSEDED:Lcom/android/server/vibrator/VibrationSession$Status;
 
-- 
2.51.1

